pkg/tfbridge/provider.go

```diff
@@ -408,9 +417,29 @@ func (p *Provider) CheckConfig(ctx context.Context, req *pulumirpc.CheckRequest)
 	// See pulumi/pulumi-terraform-bridge#1087
 	if !news.ContainsUnknowns() {
 		if err := p.preConfigureCallback(ctx, news, config); err != nil {
+			if failureErr, ok := err.(CheckFailureError); ok {
+				return &pulumirpc.CheckResponse{
+					Failures: []*pulumirpc.CheckFailure{
```

Comment:
This is reasonable! Can we make it so that the user code can return more than 1 CheckFailure, like a slice of them? This is useful to say anchor error messages on the "region" property and something else.